packages/react/src/reducer.ts

```diff
@@ -1,58 +1,49 @@
-import { IdTokenClaims } from '@logto/client';
-
 import { AuthState } from './auth-state';
 
-type Action =
-  | {
-      type: 'INITIALIZE';
-      payload: { isAuthenticated: boolean; isLoading: boolean; claims?: IdTokenClaims };
-    }
-  | { type: 'LOGIN_WITH_REDIRECT' }
-  | { type: 'HANDLE_CALLBACK_REQUEST' }
-  | { type: 'HANDLE_CALLBACK_SUCCESS'; payload: { claims: IdTokenClaims } }
-  | { type: 'LOGOUT' }
-  | { type: 'ERROR'; payload: { error: unknown } };
-
-export const reducer = (state: AuthState, action: Action): AuthState => {
-  if (action.type === 'INITIALIZE') {
-    const { isAuthenticated, isLoading, claims } = action.payload;
-
-    return {
-      ...state,
-      isInitialized: true,
-      isAuthenticated,
-      isLoading,
-      claims,
-    };
-  }
+type Action = {
+  type:
+    | 'INITIALIZE'
+    | 'LOGIN_WITH_REDIRECT'
+    | 'HANDLE_CALLBACK_REQUEST'
+    | 'HANDLE_CALLBACK_SUCCESS'
+    | 'LOGOUT'
+    | 'ERROR';
+  payload?: Partial<AuthState>;
```

Comment:
The original type anotation is more specific?

Reply:
Yes, it's more specific, but using the `no-unnecessary-condition` lint rule, the linter complained about the last `if condition` being `unnecessary`, thus I refactored the code to use switch-case instead.
But later the linter complained about lexical declarations in switch case (enabled by `no-case-declarations` rule), so I have to bring those declarations to the very top of the function to make the linter happy.

So I have to use a more generic type definition to make the variable declarations work at top level.

If we prefer using more specific type definitions for each `case`, we'll probably have to define several new `payload types` for each case, which I think is a bit off the track and unnecessary. So finally I came up with the idea of using a more generic type definition for `payload`.

Reply:
Well, I decided to revert the change but instead add a ignore for that last if-condition to avoid unnecessary headache